src/plugins/workspace/server/permission_control/client.ts

```diff
@@ -48,11 +53,46 @@ export class SavedObjectsPermissionControl {
     this.logger = logger;
   }
 
+  private generateSavedObjectKey = ({ type, id }: { type: string; id: string }) => {
+    return `${type}${id}`;
```

Comment:
```suggestion
    return `${type}:${id}`;
```
Add a separator would be nice to avoid collision.